packages/suite/src/views/wallet/coinmarket/DCA/index.tsx

```diff
@@ -0,0 +1,189 @@
+import React from 'react';
+import { Column, Divider, H2, Icon, Image, Paragraph, Row } from '@trezor/components';
+import { UseCoinmarketProps } from 'src/types/coinmarket/coinmarket';
+import { withSelectedAccountLoaded } from 'src/components/wallet';
+import { CoinmarketLayout } from '../common';
+import styled from 'styled-components';
+import { borders, colorVariants, palette, spacings } from '@trezor/theme';
+import { Translation, TrezorLink } from 'src/components/suite';
+import { COINMARKET_DOWNLOAD_INVITY_APP_URL } from '@trezor/urls';
+import { useSelector } from 'src/hooks/suite';
+
+const IconWrapper = styled.div`
+    display: flex;
+    flex-shrink: 0;
+    justify-content: center;
+    align-items: center;
+    width: 40px;
+    height: 40px;
+    border-radius: 50%;
+    background: ${palette.lightSecondaryLime500};
+    margin-right: ${spacings.sm}px;
+`;
+
+const Subheading = styled(Paragraph)`
+    color: ${palette.lightSecondaryLime500};
+`;
+
+const WrappedText = styled(Paragraph)`
+    max-width: 200px;
+    text-align: center;
+`;
+
+const Column1 = styled(Column)`
+    flex: 1 0 50%;
+    align-items: start;
+    gap: ${spacings.xxl}px;
+    background-color: ${colorVariants.standard.backgroundPrimaryDefault};
+    padding: ${spacings.xxl}px;
+    border-radius: ${borders.radii.md};
+    color: ${colorVariants.standard.textOnPrimary};
+`;
+
+const Column2 = styled(Column)`
+    flex: 1 0 50%;
+    gap: ${spacings.md}px;
+    padding: ${spacings.xxl}px;
+`;
+
+const Layout = styled(CoinmarketLayout)`
+    & [class^='CoinmarketLayout__Content'] {
+        padding: 12px;
+    }
+`;
+
+const StoreBadge = styled(Image)<{ $isLight: boolean }>`
+    ${({ $isLight }) => $isLight && `filter: invert(1);`}
+    transition: opacity 0.3s;
+    cursor: pointer;
+
+    &:hover {
+        opacity: 0.6;
+    }
+`;
+
+const StoreSeparatorWrapper = styled.div`
+    height: 26px;
+`;
+
+const DCALanding = (props: UseCoinmarketProps) => {
+    const currentTheme = useSelector(state => state.suite.settings.theme.variant);
+
+    return (
```

Comment:
What about divide these JSX into multiple smaller components? You can keep them locally in this file, but it would be more readable IMHO.

Some compositions are repeating so these could be easily refactored.